lib/http.js

```diff
@@ -4,15 +4,26 @@ var cookies = require('./cookies');
 var Q = require('q');
 var AuthApiError = require('./errors/AuthApiError');
 var config = require('./config');
+var storageBuilder = require('./storageBuilder');
+
+var httpCache = storageBuilder(localStorage, config.CACHE_STORAGE_NAME);
 
 function httpRequest(sdk, options) {
   options = options || {};
   var url = options.url,
       method = options.method,
       args = options.args,
-      dontSaveResponse = options.dontSaveResponse,
+      cacheState = options.cacheState,
       accessToken = options.accessToken;
 
+  if (options.responseCacheDuration) {
+    var cacheContents = httpCache.getStorage();
+    var cachedResponse = cacheContents[url];
+    if (cachedResponse && Date.now()/1000 < cachedResponse.expiresAt) {
```

Comment:
Can we simplify and remove the math (i.e. just use Date.now() and work in ms) since we control both ends and don't plan on exposing this outwards?